alchemist-physics/src/main/kotlin/it/unibo/alchemist/model/implementations/reactions/PhysicsUpdate.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2010-2022, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.reactions
+
+import it.unibo.alchemist.model.interfaces.Action
+import it.unibo.alchemist.model.interfaces.Condition
+import it.unibo.alchemist.model.interfaces.Dependency
+import it.unibo.alchemist.model.interfaces.Environment
+import it.unibo.alchemist.model.interfaces.GlobalReaction
+import it.unibo.alchemist.model.interfaces.Time
+import it.unibo.alchemist.model.interfaces.TimeDistribution
+import it.unibo.alchemist.model.interfaces.environments.Dynamics2DEnvironment
+import org.danilopianini.util.ImmutableListSet
+import org.danilopianini.util.ListSet
+
+/**
+ * A global Reaction responsible for updating the physics of an [Dynamics2DEnvironment].
+ */
+class PhysicsUpdate<T>(
+    /**
+     * The environment to update.
+     */
+    val environment: Dynamics2DEnvironment<T>,
+    override val timeDistribution: TimeDistribution<T>,
+) : GlobalReaction<T> {
+
+    override val outboundDependencies: ListSet<out Dependency> = ListSet.of()
+        get() = ImmutableListSet.copyOf(field)
+
+    override val inboundDependencies: ListSet<out Dependency> = ListSet.of(Dependency.MOVEMENT)
+        get() = ImmutableListSet.copyOf(field)
```

Comment:
@DanySK, I'm a bit stuck sorting out how to properly manage the Dependency Graph.
In `PhysicalBlendedSteering`, I've added an outbound dependency on `MOVEMENT`, 
I'm I missing something? Because what I'm getting is that the local reactions are executed just once.

Reply:
If you can only move as a result of the action, then it is correct. If your reaction (or its conditions) *depends on* the movement of other nodes (namely, the reaction reads the position of others, and may be scheduled sooner or later depending on their position), then they must also declare an inbound reaction.

Reply:
So, since the `GlobalReaction` reads the velocities of all the nodes in the environment, which they have been set by the local reactions, it should be:
* **Global reaction** with both an inbound and outbound MOVEMENT dependency
* **Local reaction** with just the outbound MOVEMENT dependency
Right?

Reply:
No, still the Local reaction are executed once at the start of the simulation and never again. 
Also, the time remains still at `0.0` while the step counter increases.

Reply:
GlobalReaction is an interface. It has no context, it is always implicitly treated as GLOBAL, as LOCAL and NEIGHBORHOOD make no sense.

Reply:
It's scheduled as ASAP (rate to infinity) and dependencies say that it is executable, so it gets rescheduled continually.

Reply:
> GlobalReaction is an interface. It has no context, it is always implicitly treated as GLOBAL, as LOCAL and NEIGHBORHOOD make no sense.

Yes sorry, with Global reaction i meant its implementation which is `PhysicsUpdate` and with local reaction i meant `PhysicalBlendedSteering`. 

I have already removed the context properties from the `GlobalReaction` interface. 

It's getting a little tricky for me... From what I'm getting now, having said that `PhysicsUpdate` has both in and out MOVEMENT dependency, in order to get all the `PhysicalBlendedSteering` to be executed, they should also have they inbound dependency. 

Did i miss something for the `PhysicsUpdate` having INFINITY rate? I did give it a time distribution.

Reply:
The rate is assigned by the incarnation, and Protelis defaults to "ASAP", I think. You should use a `DiracComb`. Also, I'm not sure the correct dependency is `MOVEMENT`. Can we discuss this briefly in person?